lib/msf/core/web_services/framework_extension.rb

```diff
@@ -26,6 +26,10 @@ module Helpers
       def framework
         settings.framework
       end
+
+      def get_db
+        framework.db
+      end
```

Comment:
This overrides the default behavior here:

https://github.com/rapid7/metasploit-framework/blob/6549c12473a578a055a8fa27a09a96ee0b8ade08/lib/msf/core/web_services/servlet_helper.rb#L84-L86

Importantly this change treats framework's db as the source of truth. Without this change, two database connections were being opened separately from each other. The first from the servlet helper to create the db proxy instance, and a second from framework itself